src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,68 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static DateTimeFormatter DATE_TIME_FORMATTER
+            = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private String[] report;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        String[] resultReport = new String[names.length];
+        report = new String[names.length];
+
+        LocalDate dateStart = LocalDate.parse(dateFrom, DATE_TIME_FORMATTER);
+        LocalDate dateEnd = LocalDate.parse(dateTo, DATE_TIME_FORMATTER);
+
+        StringBuilder stringBuilderFull = new StringBuilder();
+        StringBuilder stringBuilderEmpty = new StringBuilder();
+        StringBuilder builder = new StringBuilder();
+
+        stringBuilderFull.append("Report for period " + dateFrom + " - " + dateTo);
+        stringBuilderEmpty.append("Report for period " + dateFrom + " - " + dateTo);
```

Comment:
```suggestion
        StringBuilder result = new StringBuilder().append("Report for period ")
                .append(dateFrom).append(" - ").append(dateTo);
```
You need only one StringBuilder